sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -65,7 +64,11 @@ case class CreateTableLikeCommand(
         s"Source table in CREATE TABLE LIKE does not exist: '$sourceTable'")
     }
 
-    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
+    val sourceTableDesc = if (catalog.isTemporaryTable(sourceTable)) {
+      catalog.getTempViewMetadata(sourceTable.table)
```

Comment:
this is not atomic, we should have a method that return temp view metadata or None.